Source/WebCore/page/ResizeObserver.cpp

```diff
@@ -141,6 +144,10 @@ void ResizeObserver::deliverObservations()
     m_activeObservations.clear();
     auto activeObservationTargets = std::exchange(m_activeObservationTargets, { });
 
+    auto scopeExit = makeScopeExit([&] {
```

Comment:
Not sure if I understand `makeScopeExit`. It seems that it will clear `m_targetsWaitingForFirstObservation` once `scopeExit` is destroyed, which I guess it's at the end of the function? But then `m_targetsWaitingForFirstObservation` will be cleared after `m_callback->handleEvent`, which seems bad, because the callbacks could be observing new elements that will be appended into `m_targetsWaitingForFirstObservation` and shouldn't be cleared yet.

Reply:
Hmm, yes, that is possible.

Reply:
Why don't we std::exchange it to a local variable instead? Then the list still keeps object alive during the duration of this function, and it goes away at the end of function.
```cpp
targetsWaitingForFirstObservation = std::exchange(m_targetsWaitingForFirstObservation);
```

Reply:
Yes, like for `IntersectionObserver` https://github.com/WebKit/WebKit/blob/fc0d8fbdae064b740a609102f1c75f9ffee73887/Source/WebCore/page/IntersectionObserver.cpp#L291

Reply:
Done